### DISCUSSION TITLE
More elegant way to move to next workspace?

### DISCUSSION THREAD

**Discussion by nkaradzhov** [nikitabobko/AeroSpace#674] [questions-and-answers] [answered]
Hello there!

I am trying to make shortcuts to go to next and previous workspace. I only want to cycle over nonempty workspaces

`workspace --wrap-around next` takes stdin, so in theory supplying `list-workspaces --monitor focused --empty no` should give it only nonempty workspaces to cycle through

but the following config doesnt compile:
`alt-n = 'list-workspaces --monitor focused --empty no | workspace --wrap-around next'`

I have to `use exec-and-forget`, so this works:
`alt-n = 'exec-and-forget aerospace list-workspaces --monitor focused --empty no | aerospace workspace --wrap-around next'`

My question is - is there a more elegant way to achieve this without using `exec-and-forget` ?

**Comment by nikitabobko** [comment, marked as answer]
Not possible yet. Planned to be supported in the following way:

```
alt-n = 'list-workspaces --monitor focused --empty no | workspace --wrap-around next'
```

Please follow #278

**Reply by whaley** [reply]
Should this also work for `move-node-to-workspace prev|nextg`?
